app/models/tweet.rb

```diff
@@ -0,0 +1,51 @@
+class Tweet
+  attr_reader :thing
+
+  class << self
+    def for thing
+      new thing
+    end
+  end
+
+  def initialize thing
+    @thing = thing
+  end
+
+  def publish
+    twitter_client.update_with_media(content, image_file)
+  end
+
+  private
+
+  def content
+    <<~TWEET_TEXT
+      #{thing.class.name}:
+      #{thing.name}
+
+      #{thing_url}
+    TWEET_TEXT
+  end
+
+  def image_url
+    return thing.image.url if thing.image.is_a? ActiveStorage::Attached::One
+
+    thing.image
+  end
+
+  def image_file
+    Down.download image_url
+  end
+
+  def thing_url
+    "https://cwc.im#{thing.path}"
+  end
+
+  def twitter_client
```

Comment:
Later we may want to extract the twitter client to a class if we want to automatize the syndication of article on twitter, but it's really clean!

Reply:
That's a great idea, @just1602. Let's definitely do that.

It's one of the oldest issues even!
https://github.com/crimethinc/website/issues/47